src/main/java/org/jabref/logic/exporter/AtomicFileOutputStream.java

```diff
@@ -174,46 +204,44 @@ public void close() throws IOException {
             }
             super.close();
 
-            // We successfully wrote everything to the temporary file, lets copy it to the correct place
-            // First, make backup of original file and try to save file permissions to restore them later (by default: 664)
-            Set<PosixFilePermission> oldFilePermissions = EnumSet.of(PosixFilePermission.OWNER_READ,
-                    PosixFilePermission.OWNER_WRITE,
-                    PosixFilePermission.GROUP_READ,
-                    PosixFilePermission.GROUP_WRITE,
-                    PosixFilePermission.OTHERS_READ);
-            if (Files.exists(targetFile)) {
-                Files.copy(targetFile, backupFile, StandardCopyOption.REPLACE_EXISTING);
-                if (FileUtil.IS_POSIX_COMPILANT) {
-                    try {
-                        oldFilePermissions = Files.getPosixFilePermissions(targetFile);
-                    } catch (IOException exception) {
-                        LOGGER.warn("Error getting file permissions for file {}.", targetFile, exception);
-                    }
-                }
-            }
-
-            // Move temporary file (replace original if it exists)
-            Files.move(temporaryFile, targetFile, StandardCopyOption.ATOMIC_MOVE, StandardCopyOption.REPLACE_EXISTING);
-
-            // Restore file permissions
-            if (FileUtil.IS_POSIX_COMPILANT) {
-                try {
-                    Files.setPosixFilePermissions(targetFile, oldFilePermissions);
-                } catch (IOException exception) {
-                    LOGGER.warn("Error writing file permissions to file {}.", targetFile, exception);
-                }
+            if (!errorDuringWrite) {
+                // We successfully wrote everything to the temporary file, let's copy it to the correct place
+                replaceOriginalFileByWrittenFile();
             }
 
             if (!keepBackup) {
-                // Remove backup file
                 Files.deleteIfExists(backupFile);
             }
         } finally {
-            // Remove temporary file (but not the backup!)
             cleanup();
         }
     }
 
+    private void replaceOriginalFileByWrittenFile() throws IOException {
+        // Move temporary file (replace original if it exists)
+        // We implement the move as write into the original and delete the temporary one to keep file permissions etc.
```

Comment:
Then hard-exiting JabRef during a write operation may leave the file corrupt and kind of defeats the purpose of the atomic writer.

Reply:
Note to self: I'll create an ADR. - The intended implementation fixes https://github.com/JabRef/jabref/issues/8887.